2007-03-17  Jakub Jelinek  <jakub@redhat.com>

	* sysdeps/powerpc/powerpc32/power5+/fpu/s_round.S: Use GLIBC_2_1
	instead of GLIBC_2_0.
	* sysdeps/powerpc/powerpc32/power5+/fpu/s_trunc.S: Likewise.
	* sysdeps/powerpc/powerpc64/power5+/fpu/s_round.S: Likewise.
	* sysdeps/powerpc/powerpc64/power5+/fpu/s_trunc.S: Likewise.
	* sysdeps/powerpc/powerpc32/powerpc64/fpu/s_llrint.S: Include
	math_ldbl_opt.h.
	(llrintl): Add @GLIBC_2.1 compat symbol.

--- libc/powerpc-cpu/sysdeps/powerpc/powerpc32/power5+/fpu/s_round.S	15 Sep 2006 14:40:36 -0000	1.1.2.1
+++ libc/powerpc-cpu/sysdeps/powerpc/powerpc32/power5+/fpu/s_round.S	17 Mar 2007 21:52:54 -0000	1.1.2.2
@@ -32,6 +32,6 @@ weak_alias (__round, round)
 weak_alias (__round, roundl)
 strong_alias (__round, __roundl)
 #endif
-#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_0)
-compat_symbol (libm, __round, roundl, GLIBC_2_0)
+#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_1)
+compat_symbol (libm, __round, roundl, GLIBC_2_1)
 #endif
--- libc/powerpc-cpu/sysdeps/powerpc/powerpc32/power5+/fpu/s_trunc.S	15 Sep 2006 14:40:36 -0000	1.1.2.1
+++ libc/powerpc-cpu/sysdeps/powerpc/powerpc32/power5+/fpu/s_trunc.S	17 Mar 2007 21:52:54 -0000	1.1.2.2
@@ -32,6 +32,6 @@ weak_alias (__trunc, trunc)
 weak_alias (__trunc, truncl)
 strong_alias (__trunc, __truncl)
 #endif
-#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_0)
-compat_symbol (libm, __trunc, truncl, GLIBC_2_0)
+#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_1)
+compat_symbol (libm, __trunc, truncl, GLIBC_2_1)
 #endif
--- libc/powerpc-cpu/sysdeps/powerpc/powerpc32/powerpc64/fpu/s_llrint.S	15 Sep 2006 14:40:37 -0000	1.1.2.1
+++ libc/powerpc-cpu/sysdeps/powerpc/powerpc32/powerpc64/fpu/s_llrint.S	17 Mar 2007 21:52:55 -0000	1.1.2.2
@@ -18,6 +18,7 @@
    02110-1301 USA.  */
 
 #include <sysdep.h>
+#include <math_ldbl_opt.h>
 
 /* long long int[r3, r4] __llrint (double x[fp1])  */
 ENTRY (__llrint)	
@@ -41,3 +42,6 @@ weak_alias (__llrint, llrint)
 strong_alias (__llrint, __llrintl)
 weak_alias (__llrint, llrintl)
 #endif
+#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_1)
+compat_symbol (libm, __llrint, llrintl, GLIBC_2_1)
+#endif
--- libc/powerpc-cpu/sysdeps/powerpc/powerpc64/power5+/fpu/s_round.S	15 Sep 2006 14:40:37 -0000	1.1.2.1
+++ libc/powerpc-cpu/sysdeps/powerpc/powerpc64/power5+/fpu/s_round.S	17 Mar 2007 21:52:55 -0000	1.1.2.2
@@ -33,6 +33,6 @@ weak_alias (__round, round)
 weak_alias (__round, roundl)
 strong_alias (__round, __roundl)
 #endif
-#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_0)
-compat_symbol (libm, __round, roundl, GLIBC_2_0)
+#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_1)
+compat_symbol (libm, __round, roundl, GLIBC_2_1)
 #endif
--- libc/powerpc-cpu/sysdeps/powerpc/powerpc64/power5+/fpu/s_trunc.S	15 Sep 2006 14:40:37 -0000	1.1.2.1
+++ libc/powerpc-cpu/sysdeps/powerpc/powerpc64/power5+/fpu/s_trunc.S	17 Mar 2007 21:52:55 -0000	1.1.2.2
@@ -33,6 +33,6 @@ weak_alias (__trunc, trunc)
 weak_alias (__trunc, truncl)
 strong_alias (__trunc, __truncl)
 #endif
-#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_0)
-compat_symbol (libm, __trunc, truncl, GLIBC_2_0)
+#if LONG_DOUBLE_COMPAT(libm, GLIBC_2_1)
+compat_symbol (libm, __trunc, truncl, GLIBC_2_1)
 #endif
